src/berry_pots_app.c

```diff
@@ -0,0 +1,1778 @@
+                data->waitFrames = 0;
+                data->state72 = 0;
+                return TRUE;
+            }
+            break;
+        }
+    }
+
+    return FALSE;
+}
+
+static BOOL ov17_0220279C(BerryPotsAppData *data) {
+    switch (data->state72) {
+        case 0:
+            ov17_02203460(data, data->unk7C);
+            data->unk7E = data->unk20[data->unk7C].quantityOrYieldMaybe;
+            data->waitFrames = 0;
+            data->state72++;
+            break;
+        case 1:
+            if (data->waitFrames == 0) {
+                ov17_02203638(data);
+                data->unk7E--;
+                data->waitFrames = 8;
+            }
+            data->waitFrames--;
+            if (data->unk7E == 0) {
+                data->state72++;
+            }
+            break;
+        case 2:
+            if (data->runningTasks != 0) {
+                break;
+            }
+
+            data->waitFrames = 0;
+            data->state72 = 0;
+            return TRUE;
+    }
+
+    return FALSE;
+}
+
+static void ov17_02202850(BerryPotsAppData *data, u32 seqNo) {
+    Set2dSpriteAnimSeqNo(data->sprites[0], seqNo);
+    Sprite_ResetAnimCtrlState(data->sprites[0]);
+    FillWindowPixelBuffer(&data->windows[0], 0);
+
+    u32 stringWidth = FontID_String_GetWidth(4, data->cancelString, 0);
+    AddTextPrinterParameterized2(&data->windows[0], 4, data->cancelString, (48 - stringWidth) / 2, 0, 0xFF, MakeTextColor(1, 2, 0), NULL);
+
+    ScheduleWindowCopyToVram(&data->windows[0]);
+}
+
+static void ov17_022028B8(BerryPotsAppData *data) {
+    u8 unk7C = data->unk7C;
+    Sprite_SetPositionXY(data->sprites[1], (unk7C + 1) * 27, 96);
+    if (data->unk20[unk7C].growthStage > BERRY_POT_GROWTH_STAGE_GROWING) {
+        Set2dSpriteAnimSeqNo(data->sprites[1], 3);
+    } else {
+        Set2dSpriteAnimSeqNo(data->sprites[1], 2);
+    }
+}
+
+static void ov17_02202900(BerryPotsAppData *data, BOOL visible) {
+    Set2dSpriteVisibleFlag(data->sprites[1], visible);
+}
+
+static void ov17_02202910(BerryPotsAppData *data, BOOL active) {
+    Set2dSpriteAnimActiveFlag(data->sprites[1], active);
+
+    for (int i = 0; i < MAX_BERRY_POT; i++) {
+        if (data->unk20[i].soilSpriteMaybe != NULL && data->unk20[i].growthStage >= BERRY_POT_GROWTH_STAGE_SPROUTED) {
+            Set2dSpriteAnimActiveFlag(data->unk20[i].soilSpriteMaybe, active);
+        }
+    }
+}
+
+static void ov17_02202944(BerryPotsAppData *data, u32 a1, u8 a2, BOOL a3) {
+    Window *window = (a1 == 0) ? &data->windows[1] : &data->windows[2];
+    FillWindowPixelBuffer(window, 0xF);
+    if (a3) {
+        FillBgTilemapRect(data->bgConfig, 0, RGB_BLACK, 0, 18, 32, 6, 0);
+        DrawFrameAndWindow2(window, TRUE, 1, 13);
+    }
+    ov17_022036FC(data, a1, a2);
+    AddTextPrinterParameterized(window, 1, data->currentStatusString, 0, 0, 0xFF, NULL);
+    ScheduleWindowCopyToVram(window);
+    data->unk7A = 60;
+}
+
+#ifdef NONMATCHING
```

Comment:
Link to decomp.me so someone can figure it out in the future?

Reply:
Try this
```diff
@@ -920,7 +940,7 @@ static void ov17_022029C8(BerryPotsAppData *data, u32 a1) {
     Ov122_021E7488 sp24;
     MI_CpuFill8(&sp24, 0, sizeof(Ov122_021E7488));

-    memcpy(&sp24.unk0, &ov17_02203D60.unkC, sizeof(Ov122_021E9282));
+    sp24.unk0 = ov17_02203D60.unkC;
     sp24.unkC = data->listMenuItems[a1];
     sp24.unk10 = data->bgConfig;
     sp24.unk14 = ov17_02203E88[a1].unk0;
```

Reply:
> Link to decomp.me so someone can figure it out in the future?

This function *probably* uses an inlined memcpy, which to my knowledge can't be done on decomp.me (regardless, the old link is https://decomp.me/scratch/UpOrW)
> red031000 — 10/03/2023 11:00 AM
it won't without string.h defined
it will in the code itself but not on decomp.me
known issue

> Try this

This is what I had done originally. It's functionally equivalent, but the pattern with how the last two members are stored is wrong. This would do a load-store-load-store, but with an inlined memcpy (and what the game actually does), it's a load-load-store-store. The problem now is to figure out how to make everything else match again.

Reply:
scratch that, apparently it matches now HAHA